src/main/java/org/pentaho/mondrian/publish/PublishToServerCommand.java

```diff
@@ -330,23 +334,27 @@ private int publish(
    * @param fileName name of schema file on filesystem
    * @return Look up name from XML otherwise use file name
    */
-  private String determineDomainCatalogName(InputStream dataInputStream, String fileName) {
-    String domainId  = "";
+  private String determineDomainCatalogName( InputStream dataInputStream, String fileName )
+    throws ParserConfigurationException {
+    String domainId = "";
     final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+    factory.setFeature( XMLConstants.FEATURE_SECURE_PROCESSING, true );
+    factory.setAttribute( XMLConstants.ACCESS_EXTERNAL_DTD, "" );
+    factory.setAttribute( XMLConstants.ACCESS_EXTERNAL_SCHEMA, "" );
 
     try {
       DocumentBuilder builder = factory.newDocumentBuilder();
-      Document document = builder.parse(dataInputStream);
-      NodeList schemas = document.getElementsByTagName("Schema");
-      Node schema = schemas.item(0);
-      Node name = schema.getAttributes().getNamedItem("name");
+      Document document = builder.parse( dataInputStream );
+      NodeList schemas = document.getElementsByTagName( "Schema" );
+      Node schema = schemas.item( 0 );
+      Node name = schema.getAttributes().getNamedItem( "name" );
       domainId = name.getTextContent();
       dataInputStream.reset();
-    } catch (Exception e) {
-      LOG.fine("Problem occurred when trying to get schema name from document. Using filename instead.");
+    } catch ( Exception e ) {
```

Comment:
This existing generic exception seems to be unjustified, could use specific exception as in other places with this code block